core/guard/guard.go

```diff
@@ -0,0 +1,117 @@
+package guard
+
+import (
+	"github.com/gogo/protobuf/proto"
+	ic "github.com/libp2p/go-libp2p-core/crypto"
+	"time"
+
+	config "github.com/TRON-US/go-btfs-config"
+	"github.com/TRON-US/go-btfs/core/commands/storage"
+	"github.com/libp2p/go-libp2p-core/peer"
+	"github.com/tron-us/go-btfs-common/crypto"
+	guardPb "github.com/tron-us/go-btfs-common/protos/guard"
+)
+
+func NewContract(session *storage.FileContracts, configuration *config.Config, chunkHash string, chunkIndex int32) (*guardPb.ContractMeta, error) {
+	shard := session.ShardInfo[chunkHash]
+	guardPid, escrowPid, err := getGuardAndEscrowPid(configuration)
+	if err != nil {
+		return nil, err
+	}
+	return &guardPb.ContractMeta{
+		ContractId:       shard.ContractID,
+		RenterPid:        session.Renter.Pretty(),
+		HostPid:          shard.Receiver.Pretty(),
+		ShardHash:        chunkHash,
+		ShardIndex:       chunkIndex,
+		FileHash:         session.FileHash.KeyString(),
+		RentStart:        shard.StartTime,
+		RentEnd:          shard.StartTime.Add(shard.Length),
+		GuardPid:         guardPid.Pretty(),
+		EscrowPid:        escrowPid.Pretty(),
+		Price:            shard.Price,
+		Amount:           1000,
+		CollateralAmount: 0,
+		PayoutSchedule:   0,
+		NumPayouts:       5,
```

Comment:
put into const somewhere?

Reply:
OR add a TODO to put this into config

Reply:
confirmed that is optional in Guards. so I would leave it todo to future regulate all these values

Reply:
ok, maybe put 0 for now, otherwise it looks like it's already picked.